### PR TITLE
feat: Add v2 route mtu rendering to NetworkManager

### PR DESCRIPTION
**Pull request by TheRealFalcon**
## Proposed Commit Message

```
feat: Add v2 route mtu rendering to NetworkManager
```

## Merge type

- [x] Squash merge using "Proposed Commit Message"
- [ ] Rebase and merge unique commits. Requires commit messages per-commit each referencing the pull request number (#<PR_NUM>)

### CODE REVIEW

File: cloudinit/net/network_manager.py [hunk 1]
```diff
@@ -173,18 +173,44 @@ def _set_ip_method(self, family, subnet_type):
...
+          route: route0
+          key: mtu
+          value: 500
+
+        Create line under [ipv4] section:
+            route0_options=mtu=500
+
+        If the line already exists, then append the new key/value pair
+        """
+        if not self.config[section].get(f"{route}_options"):
```
Comment (holmanb) [inline cloudinit/net/network_manager.py:208]: We re-create the same string and reference the same dict location multiple times here - and also use nested f strings - a bit messy to grok. Assigning a couple of variables would make this a little easier to read.

compare this:
```python
        if not self.config[section].get(f"{route}_options"):
            self.config[section][f"{route}_options"] = f"{key}={value}"
        else:
            self.config[section][
                f"{route}_options"
            ] = f'{self.config[section][f"{route}_options"]},{key}={value}'
```

with this:

```python
        option_number = f"{route}_options"
        route_options = self.config[section].get(option_number)
        if not route_options:
            self.config[section][option_number] = f"{key}={value}"
        else:
            self.config[section][
                option_number
            ] = f'{route_options},{key}={value}'
```

File: cloudinit/net/network_manager.py [hunk 2]
```diff
@@ -173,18 +173,44 @@ def _set_ip_method(self, family, subnet_type):
...
+
+        Example:
+        Given:
+          section: ipv4
+          route: route0
+          key: mtu
+          value: 500
+
+        Create line under [ipv4] section:
+            route0_options=mtu=500
```
Comment (holmanb) [inline cloudinit/net/network_manager.py:204]: I had to dig a little to find docs that reference this syntax, but [I found them](https://networkmanager.pages.freedesktop.org/NetworkManager/NetworkManager/nm-settings-keyfile.html). Do you think we should add a link to the function docstring above?

**Review by holmanb — APPROVED** [review]
Thanks @TheRealFalcon, this is a great addition. I have a couple of nitpicks on the code, but feel free to take what you want and ignore the rest.

File: cloudinit/net/network_manager.py (hunk 2 above)
Comment (TheRealFalcon) [reply cloudinit/net/network_manager.py:204]: Most of this file is for generating these keyfiles. I'll add a more general reference, but I think it'd be odd to put it on this function specifically.

**Pull request merged**